### PR TITLE
Implement api data manipulation with redux

### PR DESCRIPTION
**Pull request by GordonOtieno**
**Implemented API data manipulation with redux**
- ``Used fetch api to  manipulate api data``
-  ``implemented fetch function to read data from the api using thunk function on page load.``
-  `` implemented thunk function to delete data from the api and dispatch appropriate action to redux store.``
-  ``Implemented redux  middleware usage to store.``

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const appId = localStorage.getItem(STORAGE_KEY);
+  const response = await fetch(`${BASE_URL}/apps/${appId}/books`);
+  try {
+    const books = await response.json();
+    return books;
+  } catch (error) {
+    return {};
+  }
+};
+
```
Comment (Ismailco) [inline src/api.js:39]: - [ ] Your Thunk action creator should fetch the data from the API, not the local storage, kindly implement fetch or Axios for this functionality so your app get the right data from the API

File: src/api.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    appId = await response.text();
+    localStorage.setItem(STORAGE_KEY, appId);
+  }
+};
+
+export const createBook = async (book) => {
+  const appId = localStorage.getItem(STORAGE_KEY);
+  const response = await post(`/apps/${appId}/books`, book);
+
+  return response.status === 201;
```
Comment (Ismailco) [inline src/api.js:26]: - [ ] Your Thunk action creator should add a book to the API, not the local storage, kindly implement fetch or Axios for this functionality so your app send the right data to the API

File: src/api.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+export const createApp = async () => {
+  let appId = localStorage.getItem(STORAGE_KEY);
+  if (!appId) {
+    const response = await post('/apps');
+
+    appId = await response.text();
+    localStorage.setItem(STORAGE_KEY, appId);
+  }
+};
```
Comment (Ismailco) [inline src/api.js:20]: - [ ] Your Thunk action creator should remove a book from the API, not the local storage, kindly implement fetch or Axios for this functionality so your app removes a book from the API

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @GordonOtieno, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Fixing the error produced in the console, related to local storage, this is the same issue why your netlify checks are not passing I guess

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by GordonOtieno** [comment]
Sorry @Ismailco, I am just storing  the key in the local storage in this case 'AmCRSB3JeMvElq5adOUo'  and making it available when needed and not the actual data, these data directly manipulated in the api. Please check their implementation in redux/book.js files

**Review by uwadonat — APPROVED** [review]
## Hi @GordonOtieno, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
-  descriptive pull request message ✔️
image


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**